lib/std/os.zig

```diff
@@ -3121,13 +3121,21 @@ pub fn getsockoptError(sockfd: fd_t) ConnectError!void {
     }
 }
 
-pub fn waitpid(pid: i32, flags: u32) u32 {
+
+pub const WaitpidRet = struct {
+   pid: i32,
+   status: u32
+};
+
+pub fn waitpid(pid: i32, flags: u32) waitpid_ret {
     // TODO allow implicit pointer cast from *u32 to *c_uint ?
     const Status = if (builtin.link_libc) c_uint else u32;
+    const Return = if (builtin.link_libc) c_int else usize;
     var status: Status = undefined;
     while (true) {
-        switch (errno(system.waitpid(pid, &status, flags))) {
-            0 => return @bitCast(u32, status),
+        var ret: Return = system.waitpid(pid, &status, flags);
+        switch (errno(ret)) {
+            0 => return waitpid_ret{ .pid =  if (@TypeOf(ret) == c_int) @intCast(i32,ret) else  @intCast(i32, @truncate(u32, ret)), .status = @bitCast(u32, status) },
```

Comment:
Keep it simpler, this is fine:
```suggestion
            0 => return WaitpidRet{ .pid =  @intCast(pid_t, rc), .status = @bitCast(u32, status) },
```